app/src/main/java/ch/epfl/sdp/peakar/general/SettingsActivity.java

```diff
@@ -150,11 +150,21 @@ private void disableCachingChanged(){
     }
 
     /**
-     * Change offline mode
+     * If the offline mode is off, it will prompt the user to the SettingsMapActivity, otherwise
+     * it will reconnect the app to the internet.
      */
     private void offlineModeChanged(){
-        //TODO Implement
-        Toast.makeText(this,"Setting not implemented yet !", Toast.LENGTH_SHORT).show();
+
+        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(this);
+
+        boolean offlineModeValue = prefs.getBoolean(this.getResources().getString(R.string.offline_mode_key), false);
+
+        if (offlineModeValue) {
+            Intent setIntent = new Intent(this, SettingsMapActivity.class);
+            startActivity(setIntent);
+        } else {
+            Toast.makeText(this,this.getResources().getString(R.string.offline_mode_off_toast), Toast.LENGTH_SHORT).show();
+        }
     }
 
```

Comment:
unregister listener when exiting the activity.
```suggestion
    @Override
    public void onBackPressed() {
        super.onBackPressed();
        sharedPref.unregisterOnSharedPreferenceChangeListener(listener);
    }
```